osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/infra/ws/DefaultWebServiceTemplateFactory.java

```diff
@@ -285,12 +297,25 @@ private SSLConnectionSocketFactory getSSLConnectionSocketFactory(final String ke
             throw new KeyStoreException("Key store is empty");
         }
 
-        // Setup SSL context, load trust and keystore and build the message
+        // Setup SSL context, load trust and key store and build the message
         // sender
         final SSLContext sslContext = SSLContexts.custom()
                 .loadKeyMaterial(keyStore, this.keyStorePassword.toCharArray())
                 .loadTrustMaterial(this.trustStoreFactory.getObject(), new TrustSelfSignedStrategy()).build();
 
-        return new SSLConnectionSocketFactory(sslContext, SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+        final HostnameVerifier hostnameVerifier = this.getHostnameVerifier();
+
+        return new SSLConnectionSocketFactory(sslContext, hostnameVerifier);
+    }
+
+    private HostnameVerifier getHostnameVerifier() throws GeneralSecurityException {
+        switch (this.webServiceTemplateHostnameVerificationStrategy) {
+        case ALLOW_ALL_HOSTNAMES:
+            return new NoopHostnameVerifier();
+        case BROWSER_COMPATIBLE_HOSTNAMES:
+            return new DefaultHostnameVerifier();
```

Comment:
The `DefaultHostnameVerifier` class is a more strict implementation of hostname verification. This improves the security of SOAP clients.